# Visualização de Dados - Relatório de Falhas
Dataset disponível em: https://drive.google.com/file/d/10HMt_3Wb4iFpFYU00Xstpr-TpwSr6C_V/view?usp=share_link

In [9]:
!pip install pandas pyarrow


zsh:1: command not found: pip


In [10]:
import pandas as pd
import requests
import io
import re

In [11]:
df = pd.read_csv('FALHAS_04_05_06_2024_Relatorio_1.csv')

In [12]:
df = pd.DataFrame(df)

In [13]:
df.drop(columns='Unnamed: 0', inplace=True)

In [14]:
df.columns = df.iloc[0]
df.drop(df.index[0], inplace=True)

In [15]:
columns = df.columns.tolist()
print("Columns:", columns)

Columns: ['KNR', 'MODELO', 'COR', 'MOTOR', 'ESTACAO', 'USUARIO', 'HALLE', 'FALHA', 'DATA']


In [16]:
df.drop(columns='MODELO', inplace=True)

In [17]:
df.drop_duplicates(inplace=True)

In [18]:
df['FALHA'].value_counts()

FALHA
PAINEL LATERAL 05. SUJEIRA NA PINTURA                     13089
Painel lateral 05. Sujeira na pintura                     12046
PAINEL LATERAL 04. CAROÇO / PICO                           6476
Painel lateral 04. Caroço / Pico                           5898
PAINEL LATERAL 01. AMASSADO                                5287
                                                          ...  
NÚMERO DO CHASSI NA CARROCERIA DANIFICADO (A) ¨               1
Número do chassi na carroceria Danificado (a) ¨               1
CHIC. PORTA (42/52/64D/64C/0612/0613) DANIFICADO (A) ¨        1
Fechadura da porta dianteira (42/52) Ajuste incorreto         1
PINO TUCKER 130 SOLTO(A) ¨                                    1
Name: count, Length: 4778, dtype: int64

In [19]:
df['FALHA'] = df['FALHA'].str.upper()

In [20]:
df[['PEÇA', 'FALHA_PEÇA']] = df['FALHA'].str.split(' ', expand=True, n=1)

In [21]:
df.head(20)

0            KNR   COR MOTOR   ESTACAO  USUARIO HALLE  \
1   2023-2056234  2R2R   DHS  HCEQS506  4013334  ZP7    
2   2023-2056234  2R2R   DHS  HCEQS506  4013334  ZP7    
3   2023-2056234  2R2R   DHS  PCEQS620  2058570  ZP6    
4   2023-2056234  2R2R   DHS  PCEQS999  2087715  ESPC   
5   2023-3896001  0Q0Q        HCEQS203  2043971  ZP5A   
6   2023-3896001  0Q0Q        HCEQS203  2043971  ZP5A   
7   2023-4896002  0Q0Q        PCEQS314  2061350  ZP5A   
8   2024-0576008  6UA1   CWL  PCEQS622  4004807  ZP8R   
9   2024-0576008  6UA1   CWL  PCEQS622  4004807  ZP8R   
10  2024-0576017  K2K2   CWL  PCEQS999  2087715  ESPC   
11  2024-0576017  K2K2   CWL  PCEQS999  2087715  ESPC   
12  2024-0576018  2R2R   DHS      STOL  FHQWRAP   ZP6   
13  2024-0576018  2R2R   DHS      STOL  FHQWRAP  ZP6    
14  2024-1316075  K2K2   CWL  HCEQS504  2051427   ZP7   
15  2024-1316075  K2K2   CWL  HCEQS504  2051427  ZP7    
16  2024-1316075  K2K2   CWL  HCEQS506  4004717   ZP7   
17  2024-1316075  K2K2   CWL  HCEQS506  4004717   ZP7   
18  2024-1316075  K2K2   CWL  HCEQS506  4004717   ZP7   
19  2024-1316075  K2K2   CWL  HCEQS506  4004717   ZP7   
20  2024-1316075  K2K2   CWL  HCEQS506  4004717   ZP7   

0                                               FALHA                DATA  \
1                     PARALAMA DIANTEIRO 01. AMASSADO  20/5/2024 21:31:52   
2                    PORTA TRASEIRA 04. CAROÇO / PICO  20/5/2024 21:30:23   
3                 MÓDULO DO ACC (13) PEÇA INCORRETA ¨  20/5/2024 20:04:56   
4               TAMPA TRASEIRA 05. SUJEIRA NA PINTURA  11/4/2024 12:15:40   
5              TAMPA DIANTEIRA 05. SUJEIRA NA PINTURA   5/6/2024 17:38:52   
6               TAMPA TRASEIRA 05. SUJEIRA NA PINTURA   5/6/2024 17:39:18   
7                         TETO 05. SUJEIRA NA PINTURA  28/5/2024 12:25:49   
8                                 FAROL (09) PEÇA NOK   9/4/2024 17:51:52   
9                                 FAROL (09) PEÇA NOK   9/4/2024 17:51:52   
10              TAMPA TRASEIRA 05. SUJEIRA NA PINTURA  11/4/2024 11:58:13   
11              TAMPA TRASEIRA 05. SUJEIRA NA PINTURA  11/4/2024 11:58:13   
12                       TESTE ELETRÔNICO (IBN_3) NOK  12/4/2024 17:38:05   
13                       TESTE ELETRÔNICO (IBN_3) NOK  12/4/2024 17:38:05   
14                   CONSOLE CENTRAL AJUSTE INCORRETO   4/4/2024 06:56:31   
15                   CONSOLE CENTRAL AJUSTE INCORRETO   4/4/2024 06:56:31   
16  APLIQUE DA TAMPA TRASEIRA (GRIP MOLDING) DANIF...   4/4/2024 06:39:45   
17              PAINEL LATERAL 05. SUJEIRA NA PINTURA   4/4/2024 06:42:16   
18                         PAINEL LATERAL CONTAMINADO   4/4/2024 06:29:51   
19                   PARALAMA DIANTEIRO 02. ONDULAÇÃO   4/4/2024 06:37:58   
20               PARALAMA DIANTEIRO PREP. PINTURA NOK   4/4/2024 06:25:44   

0       PEÇA                                         FALHA_PEÇA  
1   PARALAMA                             DIANTEIRO 01. AMASSADO  
2      PORTA                         TRASEIRA 04. CAROÇO / PICO  
3     MÓDULO                       DO ACC (13) PEÇA INCORRETA ¨  
4      TAMPA                    TRASEIRA 05. SUJEIRA NA PINTURA  
5      TAMPA                   DIANTEIRA 05. SUJEIRA NA PINTURA  
6      TAMPA                    TRASEIRA 05. SUJEIRA NA PINTURA  
7       TETO                             05. SUJEIRA NA PINTURA  
8      FAROL                                      (09) PEÇA NOK  
9      FAROL                                      (09) PEÇA NOK  
10     TAMPA                    TRASEIRA 05. SUJEIRA NA PINTURA  
11     TAMPA                    TRASEIRA 05. SUJEIRA NA PINTURA  
12     TESTE                             ELETRÔNICO (IBN_3) NOK  
13     TESTE                             ELETRÔNICO (IBN_3) NOK  
14   CONSOLE                           CENTRAL AJUSTE INCORRETO  
15   CONSOLE                           CENTRAL AJUSTE INCORRETO  
16   APLIQUE  DA TAMPA TRASEIRA (GRIP MOLDING) DANIFICADO (A) ¨  
17    PAINEL                     LATERAL 05. SUJEIRA NA PINTUR

In [22]:
df[df['PEÇA'] == 'MOTOR'].value_counts()

KNR           COR   MOTOR  ESTACAO   USUARIO  HALLE  FALHA                                            DATA                PEÇA   FALHA_PEÇA                               
2024-1736352  K2K2  DHS    PCEQS601  2077914  ZP62   MOTOR LIMPADOR VIDRO TRASEIRO (09) MONTAGEM NOK  23/4/2024 23:34:54  MOTOR  LIMPADOR VIDRO TRASEIRO (09) MONTAGEM NOK    2
2024-1456342  0Q0Q  DHS    PCEQS601  4002690  ZP62   MOTOR LIMPADOR PÁRA-BRISA BZD - FALTA LEITURA    9/4/2024 15:04:50   MOTOR  LIMPADOR PÁRA-BRISA BZD - FALTA LEITURA      2
2024-1736361  2R2R  DHS    PCEQS601  4006401  ZP62   MOTOR LIMPADOR VIDRO TRASEIRO (09) SOLTO(A) ¨    24/4/2024 06:21:13  MOTOR  LIMPADOR VIDRO TRASEIRO (09) SOLTO(A) ¨      2
2024-1736342  0Q0Q  DHS    PCEQS601  2077914  ZP62   MOTOR LIMPADOR VIDRO TRASEIRO (09) MONTAGEM NOK  23/4/2024 23:19:58  MOTOR  LIMPADOR VIDRO TRASEIRO (09) MONTAGEM NOK    2
2024-1746018  K2K2  DHS    PCEQS601  2075733  ZP62   MOTOR LIMPADOR VIDRO TRASEIRO (09) SOLTO(A) ¨    24/4/2024 09:38:57  MOT

In [23]:
df['MOTOR'].value_counts()

MOTOR
DHS    194018
CWL    117526
DRP     15086
CWS       692
            3
Name: count, dtype: int64

In [24]:
df['PEÇA'].value_counts()

PEÇA
PAINEL        92915
PORTA         66154
TAMPA         64037
PARALAMA      21839
TETO          19686
              ...  
18.               2
ESPUMA            2
SELETOR           2
COMUTADOR         2
SERVOMOTOR        1
Name: count, Length: 249, dtype: int64

In [25]:
df['HALLE'].unique()

array(['ZP7 ', 'ZP6 ', 'ESPC', 'ZP5A', 'ZP8R', 'ZP6', 'ZP7', 'ZP8',
       'ZP8 ', 'RUID', 'ROD', 'ROD ', 'CAB', 'CAB ', 'DKA', 'DKA ',
       'AGUA', 'BUY', 'BUY ', 'ZP61', 'ZP62', 'PROC', 'ZP5', 'ZP5 ',
       'PVC', 'PVC ', nan, '    '], dtype=object)

In [26]:
df['HALLE'] = df['HALLE'].str.split(' ').str[0]

In [27]:
df['PEÇA'].describe()

count     327325
unique       249
top       PAINEL
freq       92915
Name: PEÇA, dtype: object

In [28]:
df['FALHA_PEÇA'].describe()

count                             327325
unique                              2208
top       LATERAL 05. SUJEIRA NA PINTURA
freq                               25135
Name: FALHA_PEÇA, dtype: object

In [29]:
data_types = df.dtypes
print("Data Types:\n", data_types)

Data Types:
 0
KNR           object
COR           object
MOTOR         object
ESTACAO       object
USUARIO       object
HALLE         object
FALHA         object
DATA          object
PEÇA          object
FALHA_PEÇA    object
dtype: object


In [70]:
count_starts_with_2024 = df['KNR'].str.startswith('2024').sum()

print("Número de KNRs de 2024:", count_starts_with_2024)

Número de KNRs de 2024: 327318


In [71]:
count_starts_with_2023 = df['KNR'].str.startswith('2023').sum()

print("Número de KNRs de 2023:", count_starts_with_2023)

Número de KNRs de 2023: 7


In [30]:
!pip install -U ydata-profiling

zsh:1: command not found: pip


In [31]:
from ydata_profiling import ProfileReport
import pandas as pd

/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [32]:
# Gerando o relatório
profile = ProfileReport(df, title='Relatório de Análise Descritiva', explorative=True)

profile.to_file("relatorio_analise_descritiva_falhas.html")

Export report to file: 100%|██████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 324.24it/s]


In [33]:
# Importando as bibliotecas
import matplotlib.pyplot as plt
import seaborn as sns

In [34]:
plt.figure(figsize=(16, 8))

order = df['FALHA'].value_counts().nlargest(10).index

# Plotando o gráfico
sns.countplot(data=df, y='FALHA', order=order)
plt.title('Tipos de falha por veículo')
plt.xlabel('Quantidade de veículos')
plt.ylabel('Falhas')
plt.show()

/var/folders/tk/cffj7y_s7fx5cs8bnqgnpvn80000gn/T/ipykernel_18979/77814070.py:10: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  plt.show()


In [35]:
plt.figure(figsize=(16, 8))

order = df['MOTOR'].value_counts().nlargest(4).index

sns.countplot(data=df, y='MOTOR', order=order)
plt.title('Tipos de falha por motor')
plt.xlabel('Quantidade de falhas')
plt.ylabel('Falhas')
plt.show()

/var/folders/tk/cffj7y_s7fx5cs8bnqgnpvn80000gn/T/ipykernel_18979/762063765.py:9: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  plt.show()


In [36]:
df['KNR'].describe()

count           327325
unique           23972
top       2024-1746005
freq                94
Name: KNR, dtype: object

In [37]:
plt.figure(figsize=(16, 8))

order = df['KNR'].value_counts().nlargest(20).index

# Plotando o gráfico
sns.countplot(data=df, y='KNR', order=order)
plt.title('Quantidade de falhas por KNR')
plt.xlabel('Quantidade de falhas')
plt.ylabel('KNR')
plt.show()

/var/folders/tk/cffj7y_s7fx5cs8bnqgnpvn80000gn/T/ipykernel_18979/3696129182.py:10: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  plt.show()


In [38]:
media_de_falhas_por_knr = df['KNR'].value_counts().mean()
print("Média de falhas por KNR:", media_de_falhas_por_knr)

Média de falhas por KNR: 13.654471883864508


In [39]:
# import pandas as pd
# import matplotlib.pyplot as plt

# # Group by KNR and FALHA and count occurrences
# failure_counts = df.groupby(['KNR', 'FALHA']).size().reset_index(name='Count')

# # Sort and retain only top 5 failure types per KNR
# top_failures_per_knr = failure_counts.groupby('KNR').apply(
#     lambda x: x.sort_values('Count', ascending=False).head(5)
# ).reset_index(drop=True)

# # Pivot the data for plotting
# pivot_df = top_failures_per_knr.pivot(index='KNR', columns='FALHA', values='Count').fillna(0)

# # Get top 20 KNRs based on total number of top 5 failures
# pivot_df['Total'] = pivot_df.sum(axis=1)
# top_knr = pivot_df.sort_values('Total', ascending=False).head(20)
# top_knr.drop(columns='Total', inplace=True)

# # Plotting
# top_knr.plot(kind='bar', stacked=True, figsize=(16, 8))
# plt.title('Top 5 Types of Failures per KNR')
# plt.xlabel('KNR')
# plt.ylabel('Count of Failures')
# plt.legend(title='Type of Failure', bbox_to_anchor=(1.05, 1), loc='upper left')
# plt.show()

In [40]:
df['COR'].value_counts()

COR
K2K2    86577
2R2R    75643
0Q0Q    55588
A1A1    34322
5T5T    31912
6UA1    17185
6K6K     8951
0QA1     5034
2RA1     4988
K2A1     4205
6KA1     2920
Name: count, dtype: int64

In [41]:
df['COR'].describe()

count     327325
unique        11
top         K2K2
freq       86577
Name: COR, dtype: object

# Visualização de Dados - Tabela de Predict

In [43]:
df_predict = pd.read_csv('PREDICT_VW_M7_EC.csv', sep=';')

In [44]:
df_predict = pd.DataFrame(df_predict)

In [45]:
df_predict.head()

KNR                             MOD  TEMPO_MONT  TEMPO_BODY  \
0  2024-0316012  T-Cross 1.0  TPLUS  128cv  A6F         431       873.0   
1  2024-0316022  T-Cross 1.0  TPLUS  128cv  A6F         431       843.0   
2  2024-0316030  T-Cross 1.4  HIGH   150cv  A6F         822      1485.0   
3  2024-0316046  T-Cross 1.0  COMFO  85KW   A6F         435      1093.0   
4  2024-0316047  T-Cross 1.0  COMFO  128cv  A6F         432       610.0   

   TEMPO_ZP5  TEMPO_ZP6  TEMPO_ZP7  TEMPO_CABINES  TEMPO_TOTAL  DIV_MIN_ZP7  \
0       11.0      52653          0            0.0        55775            0   
1       13.0      52327        554            0.0        57376          103   
2       12.0        532          0            0.0        56161            0   
3        9.0        504          0            0.0        56717            0   
4       12.0      52326          0            0.0        55974            0   

   ...  MAX_TORQUE_3107  MAX_TORQUE_3108  MAX_TORQUE_4224  MIN_TORQUE_1302  \
0  ...            60.21            60.72            18.45            30.13   
1  ...            60.43            60.57            18.11            30.17   
2  ...            60.44            60.51            18.30            30.10   
3  ...            60.60            60.50            18.05            29.15   
4  ...            60.34            60.31            18.33            30.15   

   MIN_TORQUE_3141  MIN_TORQUE_3214  MIN_TORQUE_4204  MIN_TORQUE_4212  \
0           130.53            95.76            23.31            23.03   
1           116.84            98.58            23.07            23.10   
2           134.39            96.00            23.07            23.17   
3           142.75            94.77            23.17            23.04   
4           140.38            97.38            23.26            23.05   

   MIN_TORQUE_5947  TOTAL_TORQUE_NOK  
0            57.65                 0  
1            58.57                 0  
2            53.21                 0  
3            58.61                 1  
4            62.70                 0  

[5 rows x 80 columns]

In [46]:
df_predict.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35395 entries, 0 to 35394
Data columns (total 80 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   KNR               35395 non-null  object 
 1   MOD               35395 non-null  object 
 2   TEMPO_MONT        35395 non-null  int64  
 3   TEMPO_BODY        35390 non-null  float64
 4   TEMPO_ZP5         35159 non-null  float64
 5   TEMPO_ZP6         35395 non-null  int64  
 6   TEMPO_ZP7         35395 non-null  int64  
 7   TEMPO_CABINES     35292 non-null  float64
 8   TEMPO_TOTAL       35395 non-null  int64  
 9   DIV_MIN_ZP7       35395 non-null  int64  
 10  DIV_MIN_OPTIMO    35395 non-null  int64  
 11  DIV_MIN_PINT      35395 non-null  int64  
 12  N_POS_ZP7         35395 non-null  int64  
 13  N_ZP5             35395 non-null  int64  
 14  N_ZP5A            35395 non-null  int64  
 15  N_ZP6             35395 non-null  int64  
 16  N_CABINES         35395 non-null  int64 

In [47]:
df_predict.describe()

TEMPO_MONT    TEMPO_BODY     TEMPO_ZP5     TEMPO_ZP6     TEMPO_ZP7  \
count  35395.000000  35390.000000  35159.000000  35395.000000  35395.000000   
mean    1308.970504   2241.530913     44.024631    290.952027     81.409775   
std     4155.782300   5905.237256    777.945324   2003.325811    620.141719   
min      302.000000      2.000000  -7103.000000   -624.000000      0.000000   
25%      434.000000    609.000000      9.000000     39.000000      0.000000   
50%      465.000000    961.000000     11.000000     43.000000      0.000000   
75%      826.000000   1479.000000     14.000000     99.000000      0.000000   
max    52821.000000  99938.000000  52404.000000  57006.000000  26065.000000   

       TEMPO_CABINES    TEMPO_TOTAL   DIV_MIN_ZP7  DIV_MIN_OPTIMO  \
count   35292.000000   35395.000000  35395.000000    35395.000000   
mean        7.553043    8608.354796     15.639723       86.947026   
std       323.048368   10239.763440    479.363022      227.232713   
min         0.000000    1711.000000      0.000000        0.000000   
25%         0.000000    3813.500000      0.000000        8.000000   
50%         0.000000    6491.000000      0.000000       19.000000   
75%         0.000000    8664.000000      0.000000       64.000000   
max     52412.000000  136002.000000  55515.000000     8653.000000   

       DIV_MIN_PINT  ...  MAX_TORQUE_3107  MAX_TORQUE_3108  MAX_TORQUE_4224  \
count  35395.000000  ...     35395.000000     35395.000000     35395.000000   
mean       5.517248  ...        60.435352        60.432727        18.259367   
std       10.244340  ...         0.223471         0.239071         0.182572   
min        0.000000  ...        37.290000        34.520000         0.000000   
25%        2.000000  ...        60.310000        60.310000        18.170000   
50%        3.000000  ...        60.430000        60.430000        18.260000   
75%        6.000000  ...        60.560000        60.550000        18.340000   
max      167.000000  ...        66.080000        65.330000        18.720000   

       MIN_TORQUE_1302  MIN_TORQUE_3141  MIN_TORQUE_3214  MIN_TORQUE_4204  \
count     35395.000000     35395.000000     35395.000000     35395.000000   
mean         29.879111       146.350126        92.716723        22.945732   
std           2.208267        20.557274         4.919054         1.263938   
min           0.000000         0.000000        -0.340000         0.000000   
25%          30.090000       136.280000        90.090000        23.040000   
50%          30.150000       142.420000        92.420000        23.090000   
75%          30.200000       153.570000        95.050000        23.150000   
max          30.460000       251.470000       124.390000        27.660000   

       MIN_TORQUE_4212  MIN_TORQUE_5947  TOTAL_TORQUE_NOK  
count     35395.000000     35395.000000      35395.000000  
mean         22.891910        54.942789          1.160983  
std           1.447745         8.749900          1.265160  
min           0.000000         0.000000          0.000000  
25%          23.040000        54.170000          0.000000  
50%          23.090000        56.180000          1.000000  
75%          23.150000        58.050000          2.000000  
max          28.010000        66.620000         55.000000  

[8 rows x 78 columns]

In [48]:
df_predict["MOD"].describe()

count                              35395
unique                                13
top       T-Cross 1.4  HIGH   150cv  A6F
freq                               10676
Name: MOD, dtype: object

## Descrição da Coluna MOD

Exemplo: T-Cross 1.4 HIGH 150cv A6F

* Modelo: T-Cross
* Cilindradas: 1.4
* Tipo: Highline
* Potência: 150 [pode estar em cavalos(cv), quilowatts (KW) ou Pferdestärke (PS - Força de cavalo em alemão)]
* Transmissão: A6F - Automática de 6 velocidades ou M5F - Manual de 5 Velocidades

In [49]:
df_predict["MOD"].value_counts()

MOD
T-Cross 1.4  HIGH   150cv  A6F    10676
T-Cross 1.0  TPLUS  128cv  A6F     7992
T-Cross 1.0  COMFO  128cv  A6F     7259
T-Cross 1.0  SENSE  128PS  A6F     5138
T-Cross 1.0  TREND  70KW   M5F     1023
T-Cross 1.6  COMFO  110cv  M5F      856
T-Cross 1.0  TREND  85KW   A6F      656
T-Cross 1.0  TREND  70KW   M6F      621
T-Cross 1.0  COMFO  85KW   A6F      519
T-Cross 1.0  HIGH   85KW   A6F      289
T-Cross 1.0  COMFO  70KW   M5F      128
T-Cross 1.6  HIGH   110cv  A6F      126
T-Cross 1.6  TREND  110cv  M5F      112
Name: count, dtype: int64

In [50]:
df_predict.columns

Index(['KNR', 'MOD', 'TEMPO_MONT', 'TEMPO_BODY', 'TEMPO_ZP5', 'TEMPO_ZP6',
       'TEMPO_ZP7', 'TEMPO_CABINES', 'TEMPO_TOTAL', 'DIV_MIN_ZP7',
       'DIV_MIN_OPTIMO', 'DIV_MIN_PINT', 'N_POS_ZP7', 'N_ZP5', 'N_ZP5A',
       'N_ZP6', 'N_CABINES', 'N_OK_ZP5', 'N_NOK_ZP5A', 'N_OK_ZP5A',
       'N_NOK_ZP6', 'N_OK_ZP6', 'N_NOK_CABINES', 'N_OK_CABINES', 'N_NOK_ZP7',
       'TOTAL_FALHAS', 'PR_TETO', 'PR_DRIVE', 'DIV_SG_ELETRICA',
       'DIV_SG_ASSOALHO', 'DIV_SG_GERAL', 'FALHAS_ZP5', 'FALHAS_ZP5A',
       'FALHAS_CAB', 'FALHAS_ZP7', 'FALHAS_ROD', 'TROCA_PECA', 'REPINT',
       'MEAN_ANGLE_3108', 'MEAN_ANGLE_3205', 'MEAN_ANGLE_3214',
       'MEAN_ANGLE_3233', 'MEAN_ANGLE_3234', 'MEAN_ANGLE_3141',
       'MEAN_ANGLE_5243', 'MEAN_ANGLE_5244', 'MAX_ANGLE_3107',
       'MAX_ANGLE_3108', 'MAX_ANGLE_3141', 'MAX_ANGLE_3142', 'MAX_ANGLE_5639',
       'MAX_ANGLE_6501', 'MAX_ANGLE_6502', 'MIN_ANGLE_3233', 'MIN_ANGLE_3234',
       'MIN_ANGLE_3141', 'MIN_ANGLE_3142', 'MEAN_TORQUE_1316',
       'MEAN_TORQU

## Descrição do Subset de torques
Criei um subset com as colunas em relação a ANGLE e TORQUE.

No final, aparentemente, TOTAL_TORQUE_NOK significa o total de torques Not-OK. Mas ainda não sei como essa conta é feita.

In [51]:
subset_torques = df_predict.loc[:, ['MEAN_ANGLE_3108', 'MAX_ANGLE_3108', 'MIN_ANGLE_3141',
                       'MEAN_TORQUE_3108', 'MAX_TORQUE_3108', 'MIN_TORQUE_3141', 'TOTAL_TORQUE_NOK']]

In [52]:
subset_torques

MEAN_ANGLE_3108  MAX_ANGLE_3108  MIN_ANGLE_3141  MEAN_TORQUE_3108  \
0                 23.0            23.0          180.28             60.72   
1                 27.0            27.0          180.25             60.57   
2                 16.0            16.0          180.25             60.51   
3                 14.0            14.0          180.23             60.50   
4                 52.0            52.0          180.26             60.31   
...                ...             ...             ...               ...   
35390             27.0            27.0          180.30             60.26   
35391             18.0            18.0          180.24             60.35   
35392             28.0            28.0          180.30             60.87   
35393             25.0            25.0          180.31             60.42   
35394             17.0            17.0          180.30             60.69   

       MAX_TORQUE_3108  MIN_TORQUE_3141  TOTAL_TORQUE_NOK  
0                60.72           130.53                 0  
1                60.57           116.84                 0  
2                60.51           134.39                 0  
3                60.50           142.75                 1  
4                60.31           140.38                 0  
...                ...              ...               ...  
35390            60.26           140.54                 1  
35391            60.35           140.68                 2  
35392            60.87           140.29                 2  
35393            60.42           127.94                 2  
35394            60.69           145.90                 0  

[35395 rows x 7 columns]

## Descrição - Subset dos números de falha

Aqui eu separei em um subset as colunas com os números de falhas em relação a cada estação. E no final, em TOTAL_FALHAS, há uma soma de todas as falhas.

* N = Number
* NOK = Not OK
* OK = OK (auto-explicativo)
* N_NOK_ESTACAO = número de falhas Not-OK naquela estação

In [53]:
subset__numero_de_falhas = df_predict.loc[:, [
    'N_POS_ZP7', 'N_ZP5', 'N_ZP5A', 'N_ZP6', 'N_CABINES', 'N_OK_ZP5',
    'N_NOK_ZP5A', 'N_OK_ZP5A', 'N_NOK_ZP6', 'N_OK_ZP6', 'N_NOK_CABINES',
    'N_OK_CABINES', 'TOTAL_FALHAS'
]]

In [54]:
subset__numero_de_falhas.head()

N_POS_ZP7  N_ZP5  N_ZP5A  N_ZP6  N_CABINES  N_OK_ZP5  N_NOK_ZP5A  \
0          0      1       0      1          1         1           0   
1          1      1       1      0          0         1           1   
2          0      1       0      3          0         1           0   
3          1      1       0      0          0         1           0   
4          0      1       0      0          0         1           0   

   N_OK_ZP5A  N_NOK_ZP6  N_OK_ZP6  N_NOK_CABINES  N_OK_CABINES  TOTAL_FALHAS  
0          1          0         1              0             1             7  
1          1          0         1              0             1             7  
2          1          0         2              0             1             6  
3          1          0         2              0             1             6  
4          1          0         1              0             1             3

# Descrição Colunas de Troca de Peça e Repintura
Colunas interessantes para o modelo final.

* 0 = sem troca de peça e/ou repintura
* 1 = houve troca de peça e/ou repintura

In [55]:
df_predict['TROCA_PECA'].value_counts()

TROCA_PECA
0    34094
1     1301
Name: count, dtype: int64

In [56]:
df_predict['REPINT'].value_counts()

REPINT
0    31991
1     3404
Name: count, dtype: int64

## Descrição das colunas de Falhas
Aqui eu tô tentando descobrir o que significa mas aparentemente é o número total de falhas por estação, não necessariamente indicando qual é OK ou NOK.

In [57]:
df_predict['FALHAS_ZP7'].value_counts()

FALHAS_ZP7
3     6758
2     6607
4     5457
1     4933
5     3989
6     2550
0     1863
7     1528
8      874
9      458
10     183
11     107
12      44
13      21
14      17
16       3
15       2
18       1
Name: count, dtype: int64

In [58]:
df_predict['FALHAS_ROD'].value_counts()

FALHAS_ROD
0    33320
1     1820
2      238
3       15
4        2
Name: count, dtype: int64

In [59]:
subset_falhas_por_estacao = df_predict.loc[:, [
    'FALHAS_ZP5', 'FALHAS_ZP5A', 'FALHAS_CAB', 'FALHAS_ZP7', 'FALHAS_ROD',
]]

In [60]:
subset_falhas_por_estacao.head()

FALHAS_ZP5  FALHAS_ZP5A  FALHAS_CAB  FALHAS_ZP7  FALHAS_ROD
0           1            3           0           1           0
1           0            2           0           4           0
2           0            0           1           5           0
3           0            0           1           4           0
4           0            0           1           1           0

# Visualização de Dados - Tabela de STATUS

In [61]:
df_status = pd.read_csv('STATUS_PREDICTOR_ANO.csv')

/var/folders/tk/cffj7y_s7fx5cs8bnqgnpvn80000gn/T/ipykernel_18979/4242501525.py:1: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15) have mixed types. Specify dtype option on import or set low_memory=False.
  df_status = pd.read_csv('STATUS_PREDICTOR_ANO.csv')


In [63]:
df_status = pd.DataFrame(df_status)

In [65]:
df_status.head(20)

Unnamed: 0    Unnamed: 1 Unnamed: 2           Unnamed: 3 2023-4026002  \
0          NaN           KNR     STATUS                 DATA          NaN   
1          NaN  2023-2016173       G620  2023-06-06 07:50:27          NaN   
2          NaN  2023-2016173       M620  2023-06-06 07:50:27          NaN   
3          NaN  2023-2026032       G620  2023-06-20 17:35:51          NaN   
4          NaN  2023-2026032       M620  2023-06-20 17:35:51          NaN   
5          NaN  2023-2026085       G620  2023-06-22 20:22:36          NaN   
6          NaN  2023-2026085       M620  2023-06-22 20:22:36          NaN   
7          NaN  2023-2026098       G620  2023-06-23 20:40:34          NaN   
8          NaN  2023-2026098       M620  2023-06-23 20:40:34          NaN   
9          NaN  2023-2026162       G620  2023-06-28 07:31:41          NaN   
10         NaN  2023-2026162       M620  2023-06-28 07:31:41          NaN   
11         NaN  2023-2026162       G710  2023-06-28 08:16:42          NaN   
12         NaN  2023-2026175       G620  2023-06-27 13:56:38          NaN   
13         NaN  2023-2026175       M620  2023-06-27 13:56:38          NaN   
14         NaN  2023-2026215       M644  2023-06-29 10:06:13          NaN   
15         NaN  2023-2026215       G620  2023-06-29 10:31:31          NaN   
16         NaN  2023-2026215       M620  2023-06-29 10:31:31          NaN   
17         NaN  2023-2026291       G620  2023-06-22 18:23:52          NaN   
18         NaN  2023-2026291       M620  2023-06-22 18:23:52          NaN   
19         NaN  2023-2026304       G620  2023-06-26 07:04:52          NaN   

   R600 2023-09-29 07:14:44 2024-0346154 A600 2024-01-15 11:11:44  \
0   NaN                 NaN          NaN  NaN                 NaN   
1   NaN                 NaN          NaN  NaN                 NaN   
2   NaN                 NaN          NaN  NaN                 NaN   
3   NaN                 NaN          NaN  NaN                 NaN   
4   NaN                 NaN          NaN  NaN                 NaN   
5   NaN                 NaN          NaN  NaN                 NaN   
6   NaN                 NaN          NaN  NaN                 NaN   
7   NaN                 NaN          NaN  NaN                 NaN   
8   NaN                 NaN          NaN  NaN                 NaN   
9   NaN                 NaN          NaN  NaN                 NaN   
10  NaN                 NaN          NaN  NaN                 NaN   
11  NaN                 NaN          NaN  NaN                 NaN   
12  NaN                 NaN          NaN  NaN                 NaN   
13  NaN                 NaN          NaN  NaN                 NaN   
14  NaN                 NaN          NaN  NaN                 NaN   
15  NaN                 NaN          NaN  NaN                 NaN   
16  NaN                 NaN          NaN  NaN                 NaN   
17  NaN                 NaN          NaN  NaN                 NaN   
18  NaN                 NaN          NaN  NaN                 NaN   
19  NaN                 NaN          NaN  NaN                 NaN   

   2024-1556203 R800 2024-04-10 19:03:20 2024-2536233 G710 2024-06-14 17:36:53  
0           NaN  NaN                 NaN          NaN  NaN                 NaN  
1           NaN  NaN                 NaN          NaN  NaN                 NaN  
2           NaN  NaN                 NaN          NaN  NaN                 NaN  
3           NaN  NaN                 NaN          NaN  NaN                 NaN  
4           NaN  NaN                 NaN          NaN  NaN                 NaN  
5           NaN  NaN                 NaN          NaN  NaN                 NaN  
6           NaN  NaN                 NaN          NaN  NaN                 NaN  
7           NaN  NaN                 NaN          NaN  NaN                 NaN  
8           NaN  NaN                 NaN          NaN  NaN                 NaN  
9           NaN  NaN                 NaN          NaN  NaN                 NaN  
10          NaN  NaN                 NaN     

In [66]:
df_status['Unnamed: 1'].value_counts()

Unnamed: 1
2023-3896001    145
2023-3546053    116
2023-3536275    100
2023-3556166     92
2023-3536217     87
               ... 
2023-2156024      2
2023-2156023      2
2023-2156022      2
2023-2046372      1
KNR               1
Name: count, Length: 20429, dtype: int64

In [68]:
count_starts_with_2024 = df_status['Unnamed: 1'].str.startswith('2024').sum()

print("Número de KNRs de 2024:", count_starts_with_2024)

Número de KNRs de 2024: 0


# Visualização de Dados - Tabela de Resultados_04_06_2024_full

In [77]:
df_resultados = pd.read_csv('RESULTADOS_04_06_2024_full_teste.csv', compression="gzip")

In [78]:
df_resultados = pd.DataFrame(df_resultados)

In [79]:
df_resultados.head()

Unnamed: 0           KNR    NAME  ID  STATUS UNIT  VALUE_ID VALUE  \
0           0  2023-2056234  012001   1      10  NaN       NaN   NaN   
1           1  2023-2056234  012002   1      10  NaN       NaN   NaN   
2           2  2023-2056234  012003   1      10  NaN       NaN   NaN   
3           3  2023-2056234  012004   1      10  NaN       NaN   NaN   
4           4  2023-2056234  012008   1      10  NaN       NaN   NaN   

                  DATA  
0  2024-05-16 13:41:25  
1  2024-05-16 14:12:35  
2  2024-05-16 14:12:27  
3  2024-05-16 14:11:07  
4  2024-05-16 14:13:07

In [80]:
count_starts_with_2024 = df_resultados['KNR'].str.startswith('2024').sum()

print("Número de KNRs de 2024:", count_starts_with_2024)

Número de KNRs de 2024: 21673378


In [81]:
count_starts_with_2023 = df_resultados['KNR'].str.startswith('2023').sum()

print("Número de KNRs de 2024:", count_starts_with_2023)

Número de KNRs de 2024: 874
